nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -315,6 +323,21 @@ protected OkHttpClient createHttpClient(ConfigurationContext context) {
             clientBuilder.sslSocketFactory(sslContext.getSocketFactory(), trustManager);
         }
 
+        final ProxyConfiguration proxyConfig = ProxyConfiguration.getConfiguration(context);
+
+        final Proxy proxy = proxyConfig.createProxy();
+        if (!Proxy.Type.DIRECT.equals(proxy.type())) {
+            clientBuilder.proxy(proxy);
+            if (proxyConfig.hasCredential()) {
+                clientBuilder.proxyAuthenticator((route, response) -> {
+                    final String credential= Credentials.basic(proxyConfig.getProxyUserName(), proxyConfig.getProxyUserPassword());
+                    return response.request().newBuilder()
+                                   .header("Proxy-Authorization", credential)
```

Comment:
Is `Proxy-Authorization` a standard header for this use case?

Reply:
Thanks for your review @MikeThomsen!
Yes, it is a standard one: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Proxy-Authorization
We have BasicProxyAuthenticator that does the same thing, but for that, we would need to depend on `nifi-web-client`: https://github.com/apache/nifi/blob/864036674ec101c81701bbd8fbb7f96ad25dd3be/nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/BasicProxyAuthenticator.java#L27
But if we do that, it would be good to swap the direct usage of OkHttp to the StandardWebClientService. I discussed it with @exceptionfactory, and we will do that in a separate PR.